### PR TITLE
Will/healthcheck jwt auth

### PR DESCRIPTION
**Pull request by willmostly**
This is required for interoperability with the trino UI auth options now available.

### CODE REVIEW

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java [hunk 1]
```diff
@@ -159,6 +241,7 @@ private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
 
     } catch (Exception e) {
       log.error("Error parsing cluster stats from [{}]", response, e);
+      e.printStackTrace();
```
Comment (thorbjornsen) [inline gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java]: This leftover from debugging?

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java [hunk 2]
```diff
@@ -134,6 +161,61 @@ private String queryCluster(String target) {
...
+      String nodeStatsSql = "select\n"
+              + "count_if( not coordinator AND upper(state) = 'ACTIVE') as activeWorkers\n"
+              + "from system.runtime.nodes";
+      ResultSet nodeStatsResult = statement.executeQuery(nodeStatsSql);
+      nodeStatsResult.next();
+      clusterStats.setNumWorkerNodes(nodeStatsResult.getInt("activeWorkers"));
+      clusterStats.setProxyTo(backend.getProxyTo());
+      clusterStats.setExternalUrl(backend.getExternalUrl());
+      clusterStats.setRoutingGroup(backend.getRoutingGroup());
+      clusterStats.setHealthy(true);
```
Comment (thorbjornsen) [inline gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java:210]: Not sure if I'm following... this sets healthy to true, but then the scope exits and sets healthy to false

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java [hunk 3]
```diff
@@ -134,6 +161,61 @@ private String queryCluster(String target) {
...
+    } catch (SQLException e) {
+      log.error("Error querying cluster stats: " + e);
+    }
+    clusterStats.setHealthy(false);
```
Comment (thorbjornsen) [inline gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java:215]: Intended to set it to false here no matter what?

Comment (thorbjornsen) [reply gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java:215]: Maybe this is intended to be inside the catch

**Review by thorbjornsen — CHANGES_REQUESTED** [review]
Quick once over. Couple of issues

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java (hunk 1 above)
Comment (willmostly) [reply gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java]: yep

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java (hunk 2 above)
Comment (willmostly) [reply gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java:210]: Looks like I forgot a return statement

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java (hunk 3 above)
Comment (willmostly) [reply gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java:215]: same as above - forgot a return statement in the try block, so this should only be reached if the health check fails.

File: gateway-ha/src/main/resources/gateway-ha-persistence.sql [hunk 4]
```diff
@@ -75,4 +75,14 @@ CREATE TABLE IF NOT EXISTS exact_match_source_selectors (
 
     PRIMARY KEY (environment, source, query_type),
     UNIQUE (source, environment, query_type, resource_group_id)
-);
\ No newline at end of file
+);
+
```
Comment (thorbjornsen) [inline gateway-ha/src/main/resources/gateway-ha-persistence.sql]: Just curious if/how these database entries get managed. Any chance these could build up over time and cause an issue?

**Comment by thorbjornsen** [comment]
LGTM

**Pull request merged**